_posts/2020-01-01-2019-summary.md

```diff
@@ -0,0 +1,178 @@
+any corporate interest, for years to come.
+
+Since then, the community around Presto has grown and consolidated. We’ve seen contributions 
+from more than 120 people across over 20 companies. Every week, 280 users and developers 
+interact in the project’s [Slack channel](/slack.html).
+
+With the collaboration of companies such as [Starburst](https://starburstdata.com), [Qubole](https://qubole.com), 
+[Varada](https://varada.io), [Twitter](https://twitter.com), [ARM Treasure Data](https://www.treasuredata.com),
+[Wix](https://wix.com), [Redhat](https://www.redhat.com), and the [Big Things community](https://www.meetup.com/Big-things-are-happening-here/),
+we ran several Presto summits across the world ([Tel Aviv]({% post_url 2019-05-03-Presto-Conference-Israel %}),
+[San Francisco]({% post_url 2019-05-17-Presto-Summit %}), [Tokyo]({% post_url 2019-07-11-report-for-presto-conference-tokyo %}), 
+[Bangalore]({% post_url 2019-09-05-Presto-Summit-Bangalore %}) and New York) which 
+brought together thousands of members from the larger Presto community.
+
+The project has been more active than ever. We completed 28 releases comprised of more than 2850 
+commits in over 1500 pull requests. Of course, that alone is not a good measure of progress, so 
+let’s take a closer look at everything that went in.
+
+<!--more-->
+
+# Improvements
+
+## Language Features
+
+* [`FETCH FIRST n ROWS [ONLY] [WITH TIES]`](/docs/current/sql/select.html#limit-or-fetch-first-clauses) 
+standard syntax. The `WITH TIES` clause is particularly useful when some of the rows have the same 
+value for the columns being used to order the results of a query.
+	(TODO: sample query)
+* [`OFFSET`](/docs/current/sql/select.html#offset-clause) syntax.
+* [`COMMENT ON <table>`](/docs/current/sql/comment.html) syntax to 
+  set or remove table comments, and ability to see table comments via `DESCRIBE` or the 
+  `system.metadata.table_comments` table.
+* Support for ``LATERAL`` in the context of an outer join.
+* Support for ``UNNEST`` in the context of ``LEFT JOIN``. With this feature, it is now possible 
+  to preserve the outer row when the array contains zero elements or is NULL.
+	
+  ```sql	
+  SELECT * FROM t LEFT JOIN UNNEST(t.a) u(v) ON true
+  ```
+
+* Window functions now support the IGNORE NULLS clause. This is useful when combined with 
+  functions such as LEAD, LAG, FIRST_VALUE, LAST_VALUE and NTH_VALUE if the dataset contains nulls.
+* `.*` for ROW expressions.
+* `ROW` types now support the subscript operator to access fields by index. This improves usability 
+  when dealing with ROW type containing anonymous fields.
+  
+    ![](/assets/blog/2019-review/row-ordinal.png)
+
+* [`CREATE SCHEMA`](/docs/current/sql/create-schema.html) syntax and support 
+  in various connectors (Hive, Iceberg, MySQL, PostgreSQL, Redshift, SQL Server, Phoenix).
+
+## Query Engine
+
+* Extend support for conditional, lazy loading and processing (a.k.a., late materialization) to 
```

Comment:
* Better CBO support for projections:
https://github.com/prestosql/presto/pull/1431
https://github.com/prestosql/presto/pull/1359
* improved subquery planning (from performance point of view)
https://github.com/prestosql/presto/pull/1329
* improved reporting of physical data and network data read/sent
https://github.com/prestosql/presto/pull/326
https://github.com/prestosql/presto/pull/271
https://github.com/prestosql/presto/pull/133
* spill does not incur performance penalty when not triggered:
https://github.com/prestosql/presto/pull/315 
* spill for window and order by operators and spill productization in general:
https://github.com/prestosql/presto/pull/228